src/collision.cpp

```diff
@@ -222,74 +222,16 @@ static inline void getNeighborConnectingFace(const v3s16 &p,
 		*neighbors |= v;
 }
 
-collisionMoveResult collisionMoveSimple(Environment *env, IGameDef *gamedef,
-		f32 pos_max_d, const aabb3f &box_0,
-		f32 stepheight, f32 dtime,
-		v3f *pos_f, v3f *speed_f,
-		v3f accel_f, ActiveObject *self,
-		bool collideWithObjects)
+static bool collisionEngeInBox(const v3f &edge, const aabb3f &box)
 {
-	#define PROFILER_NAME(text) (s_env ? ("Server: " text) : ("Client: " text))
-	static bool time_notification_done = false;
-	Map *map = &env->getMap();
-	ServerEnvironment *s_env = dynamic_cast<ServerEnvironment*>(env);
-
-	ScopeProfiler sp(g_profiler, PROFILER_NAME("collisionMoveSimple()"), SPT_AVG);
-
-	collisionMoveResult result;
-
-	/*
-		Calculate new velocity
-	*/
-	if (dtime > DTIME_LIMIT) {
-		if (!time_notification_done) {
-			time_notification_done = true;
-			warningstream << "collisionMoveSimple: maximum step interval exceeded,"
-					" lost movement details!"<<std::endl;
-		}
-		dtime = DTIME_LIMIT;
-	} else {
-		time_notification_done = false;
-	}
-
-	v3f dpos_f = (*speed_f + accel_f * 0.5f * dtime) * dtime;
-	v3f newpos_f = *pos_f + dpos_f;
-	*speed_f += accel_f * dtime;
-
-	// If the object is static, there are no collisions
-	if (dpos_f == v3f())
-		return result;
-
-	// Limit speed for avoiding hangs
-	speed_f->Y = rangelim(speed_f->Y, -5000, 5000);
-	speed_f->X = rangelim(speed_f->X, -5000, 5000);
-	speed_f->Z = rangelim(speed_f->Z, -5000, 5000);
-
-	*speed_f = truncate(*speed_f, 10000.0f);
-
-	/*
-		Collect node boxes in movement range
-	*/
-	std::vector<NearbyCollisionInfo> cinfo;
-	{
-	//TimeTaker tt2("collisionMoveSimple collect boxes");
-	ScopeProfiler sp2(g_profiler, PROFILER_NAME("collision collect boxes"), SPT_AVG);
-
-	v3f minpos_f(
-		MYMIN(pos_f->X, newpos_f.X),
-		MYMIN(pos_f->Y, newpos_f.Y) + 0.01f * BS, // bias rounding, player often at +/-n.5
-		MYMIN(pos_f->Z, newpos_f.Z)
-	);
-	v3f maxpos_f(
-		MYMAX(pos_f->X, newpos_f.X),
-		MYMAX(pos_f->Y, newpos_f.Y),
-		MYMAX(pos_f->Z, newpos_f.Z)
-	);
-	v3s16 min = floatToInt(minpos_f + box_0.MinEdge, BS) - v3s16(1, 1, 1);
-	v3s16 max = floatToInt(maxpos_f + box_0.MaxEdge, BS) + v3s16(1, 1, 1);
+	return (box.MinEdge.X < edge.X) && (edge.X < box.MaxEdge.X) &&
+			(box.MinEdge.Y < edge.Y) && (edge.Y < box.MaxEdge.Y) &&
+			(box.MinEdge.Z < edge.Z) && (edge.Z < box.MaxEdge.Z);
+}
```

Comment:
reimplementation of `aabbox3d::isPointTotalInside`

Reply:
fixed